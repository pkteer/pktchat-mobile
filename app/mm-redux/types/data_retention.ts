// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export type GlobalDataRetentionPolicy = {
    file_deletion_enabled: boolean;
    file_retention_cutoff: number;
    message_deletion_enabled: boolean;
    message_retention_cutoff: number;
}

export type TeamDataRetentionPolicy = {
    post_duration: number;
    team_id?: string;
}

export type ChannelDataRetentionPolicy = {
    post_duration: number;
    channel_id?: string;
}
