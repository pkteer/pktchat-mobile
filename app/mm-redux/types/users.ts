// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import {Channel} from './channels';
import {PostType} from './posts';
import {Team} from './teams';
import {$ID, IDMappedObjects, RelationOneToMany, RelationOneToOne, Dictionary} from './utilities';
export type UserNotifyProps = {
    auto_responder_active?: 'true' | 'false';
    auto_responder_message?: string;
    desktop: 'default' | 'all' | 'mention' | 'none';
    desktop_threads?: 'all' | 'mention';
    desktop_notification_sound?: string;
    desktop_sound: 'true' | 'false';
    email: 'true' | 'false';
    email_threads?: 'all' | 'mention';
    mark_unread: 'all' | 'mention';
    push: 'default' | 'all' | 'mention' | 'none';
    push_threads?: 'all' | 'mention';
    push_status: 'ooo' | 'offline' | 'away' | 'dnd' | 'online';
    comments: 'never' | 'root' | 'any';
    first_name: 'true' | 'false';
    channel: 'true' | 'false';
    mention_keys: string;
    user_id?: string;
};
export type UserProfile = {
    id: string;
    create_at: number;
    update_at: number;
    delete_at: number;
    username: string;
    auth_data: string;
    auth_service: string;
    email: string;
    email_verified: boolean;
    nickname: string;
    first_name: string;
    last_name: string;
    position: string;
    roles: string;
    remote_id?: string;
    locale: string;
    props: Dictionary<string>;
    notify_props: UserNotifyProps;
    terms_of_service_id: string;
    terms_of_service_create_at: number;
    timezone?: UserTimezone;
    is_bot: boolean;
    last_picture_update: number;
};
export type UsersState = {
    currentUserId: string;
    isManualStatus: RelationOneToOne<UserProfile, boolean>;
    mySessions: any[];
    profiles: IDMappedObjects<UserProfile>;
    profilesInTeam: RelationOneToMany<Team, UserProfile>;
    profilesNotInTeam: RelationOneToMany<Team, UserProfile>;
    profilesWithoutTeam: Set<string>;
    profilesInChannel: RelationOneToMany<Channel, UserProfile>;
    profilesNotInChannel: RelationOneToMany<Channel, UserProfile>;
    statuses: RelationOneToOne<UserProfile, string>;
    stats: any;
};
export type UserTimezone = {
    useAutomaticTimezone: boolean | string;
    automaticTimezone: string;
    manualTimezone: string;
};
export type UserActivity = {
    [x in PostType]: {
        [y in $ID<UserProfile>]: | {
            ids: Array<$ID<UserProfile>>;
            usernames: Array<UserProfile['username']>;
        } | Array<$ID<UserProfile>>;
    };
};

export type UserStatus = {
	user_id: string;
	status: string;
	manual: boolean;
	last_activity_at: number;
	active_channel?: string;
}

export enum CustomStatusDuration {
    DONT_CLEAR = '',
    THIRTY_MINUTES = 'thirty_minutes',
    ONE_HOUR = 'one_hour',
    FOUR_HOURS = 'four_hours',
    TODAY = 'today',
    THIS_WEEK = 'this_week',
    DATE_AND_TIME = 'date_and_time',
}

export type UserCustomStatus = {
    emoji: string;
    text: string;
    expires_at?: string;
    duration: CustomStatusDuration;
}
