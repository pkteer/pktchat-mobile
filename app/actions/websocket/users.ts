// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {fetchChannelAndMyMember} from '@actions/helpers/channels';
import {loadChannelsForTeam} from '@actions/views/channel';
import {getMe} from '@actions/views/user';
import {Client4} from '@client/rest';
import {ChannelTypes, TeamTypes, UserTypes, RoleTypes} from '@mm-redux/action_types';
import {notVisibleUsersActions} from '@mm-redux/actions/helpers';
import {General} from '@mm-redux/constants';
import {getAllChannels, getCurrentChannelId, getChannelMembersInChannels} from '@mm-redux/selectors/entities/channels';
import {getCurrentTeamId} from '@mm-redux/selectors/entities/teams';
import {getCurrentUser, getCurrentUserId} from '@mm-redux/selectors/entities/users';
import {ActionResult, DispatchFunc, GenericAction, GetStateFunc, batchActions} from '@mm-redux/types/actions';
import {WebSocketMessage} from '@mm-redux/types/websocket';
import EventEmitter from '@mm-redux/utils/event_emitter';
import {isGuest} from '@mm-redux/utils/user_utils';

export function handleStatusChangedEvent(msg: WebSocketMessage): GenericAction {
    return {
        type: UserTypes.RECEIVED_STATUSES,
        data: [{user_id: msg.data.user_id, status: msg.data.status}],
    };
}

export function handleUserAddedEvent(msg: WebSocketMessage) {
    return async (dispatch: DispatchFunc, getState: GetStateFunc): Promise<ActionResult> => {
        try {
            const state = getState();
            const currentChannelId = getCurrentChannelId(state);
            const currentTeamId = getCurrentTeamId(state);
            const currentUserId = getCurrentUserId(state);
            const teamId = msg.data.team_id;
            const actions: GenericAction[] = [{
                type: ChannelTypes.CHANNEL_MEMBER_ADDED,
                data: {
                    channel_id: msg.broadcast.channel_id,
                    user_id: msg.data.user_id,
                },
            }];

            if (msg.broadcast.channel_id === currentChannelId) {
                const stat = await Client4.getChannelStats(currentChannelId);
                actions.push({
                    type: ChannelTypes.RECEIVED_CHANNEL_STATS,
                    data: stat,
                });
            }

            if (teamId === currentTeamId && msg.data.user_id === currentUserId) {
                const channelActions = await fetchChannelAndMyMember(msg.broadcast.channel_id);

                if (channelActions.length) {
                    actions.push(...channelActions);
                }
            }

            dispatch(batchActions(actions, 'BATCH_WS_USER_ADDED'));
        } catch (error) {
            //do nothing
        }
        return {data: true};
    };
}

export function handleUserRemovedEvent(msg: WebSocketMessage) {
    return async (dispatch: DispatchFunc, getState: GetStateFunc): Promise<ActionResult> => {
        try {
            const state = getState();
            const channels = getAllChannels(state);
            const currentChannelId = getCurrentChannelId(state);
            const currentTeamId = getCurrentTeamId(state);
            const currentUser = getCurrentUser(state);
            const actions: GenericAction[] = [];
            let channelId;
            let userId;

            if (msg.data.user_id) {
                userId = msg.data.user_id;
                channelId = msg.broadcast.channel_id;
            } else if (msg.broadcast.user_id) {
                channelId = msg.data.channel_id;
                userId = msg.broadcast.user_id;
            }

            if (userId) {
                actions.push({
                    type: ChannelTypes.CHANNEL_MEMBER_REMOVED,
                    data: {
                        channel_id: channelId,
                        user_id: userId,
                    },
                });
            }

            const channel = channels[currentChannelId];

            if (msg.data?.user_id !== currentUser.id) {
                const members = getChannelMembersInChannels(state);
                const isMember = Object.values(members).some((member) => member[msg.data.user_id]);
                if (channel && isGuest(currentUser.roles) && !isMember) {
                    actions.push({
                        type: UserTypes.PROFILE_NO_LONGER_VISIBLE,
                        data: {user_id: msg.data.user_id},
                    }, {
                        type: TeamTypes.REMOVE_MEMBER_FROM_TEAM,
                        data: {team_id: channel.team_id, user_id: msg.data.user_id},
                    });
                }
            }

            let redirectToDefaultChannel = false;
            if (msg.broadcast.user_id === currentUser.id && currentTeamId) {
                const {data: myData}: any = await dispatch(loadChannelsForTeam(currentTeamId, true));

                if (myData?.channels && myData?.channelMembers) {
                    actions.push({
                        type: ChannelTypes.RECEIVED_MY_CHANNELS_WITH_MEMBERS,
                        data: myData,
                    });
                }

                if (channel) {
                    actions.push({
                        type: ChannelTypes.LEAVE_CHANNEL,
                        data: {
                            id: msg.data.channel_id,
                            user_id: currentUser.id,
                            team_id: channel.team_id,
                            type: channel.type,
                        },
                    });
                }

                if (msg.data.channel_id === currentChannelId) {
                    // emit the event so the client can change his own state
                    redirectToDefaultChannel = true;
                }
                if (isGuest(currentUser.roles)) {
                    const notVisible = await notVisibleUsersActions(state);
                    if (notVisible.length) {
                        actions.push(...notVisible);
                    }
                }
            } else if (msg.data.channel_id === currentChannelId) {
                const stat = await Client4.getChannelStats(currentChannelId);
                actions.push({
                    type: ChannelTypes.RECEIVED_CHANNEL_STATS,
                    data: stat,
                });
            }

            dispatch(batchActions(actions, 'BATCH_WS_USER_REMOVED'));
            if (redirectToDefaultChannel) {
                EventEmitter.emit(General.REMOVED_FROM_CHANNEL, channel.display_name);
                EventEmitter.emit(General.SWITCH_TO_DEFAULT_CHANNEL, currentTeamId);
            }
        } catch {
            // do nothing
        }

        return {data: true};
    };
}

export function handleUserRoleUpdated(msg: WebSocketMessage) {
    return async (dispatch: DispatchFunc): Promise<ActionResult> => {
        try {
            const roles = msg.data.roles.split(' ');
            const data = await Client4.getRolesByNames(roles);

            dispatch({
                type: RoleTypes.RECEIVED_ROLES,
                data,
            });
        } catch {
            // do nothing
        }

        return {data: true};
    };
}

export function handleUserUpdatedEvent(msg: WebSocketMessage) {
    return (dispatch: DispatchFunc, getState: GetStateFunc): ActionResult => {
        const currentUser = getCurrentUser(getState());
        const user = msg.data.user;

        if (user.id === currentUser.id) {
            if (user.update_at > currentUser.update_at) {
                // Need to request me to make sure we don't override with sanitized fields from the
                // websocket event
                dispatch(getMe());
            }
        } else {
            dispatch({
                type: UserTypes.RECEIVED_PROFILES,
                data: {
                    [user.id]: user,
                },
            });
        }
        return {data: true};
    };
}
