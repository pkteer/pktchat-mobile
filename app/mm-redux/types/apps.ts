// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {UserProfile} from './users';

export type AppManifest = {
    app_id: string;
    display_name: string;
    description?: string;
    homepage_url?: string;
}

export type AppModalState = {
    form: AppForm;
    call: AppCallRequest;
}

export type AppsState = {
    bindings: AppBinding[];
    bindingsForms: AppCommandFormMap;
    threadBindings: AppBinding[];
    threadBindingsForms: AppCommandFormMap;
    threadBindingsChannelId: string;
    pluginEnabled: boolean;
};

export type AppBinding = {
    app_id: string;
    location?: string;
    icon?: string;

    // Label is the (usually short) primary text to display at the location.
    // - For LocationPostMenu is the menu item text.
    // - For LocationChannelHeader is the dropdown text.
    // - For LocationCommand is the name of the command
    label: string;

    // Hint is the secondary text to display
    // - LocationPostMenu: not used
    // - LocationChannelHeader: tooltip
    // - LocationCommand: the "Hint" line
    hint?: string;

    // Description is the (optional) extended help text, used in modals and autocomplete
    description?: string;

    role_id?: string;
    depends_on_team?: boolean;
    depends_on_channel?: boolean;
    depends_on_user?: boolean;
    depends_on_post?: boolean;

    // A Binding is either an action (makes a call), a Form, or is a
    // "container" for other locations - i.e. menu sub-items or subcommands.
    bindings?: AppBinding[];
    form?: AppForm;
    submit?: AppCall;
};

export type AppCallValues = {
    [name: string]: any;
};

export type AppCall = {
    path: string;
    expand?: AppExpand;
    state?: any;
};

export type AppCallRequest = AppCall & {
    context: AppContext;
    values?: AppCallValues;
    raw_command?: string;
    selected_field?: string;
    query?: string;
};

export type AppCallResponseType = string;

export type AppCallResponse<Res = unknown> = {
    type: AppCallResponseType;
    text?: string;
    data?: Res;
    navigate_to_url?: string;
    use_external_browser?: boolean;
    call?: AppCall;
    form?: AppForm;
    app_metadata?: AppMetadataForClient;
};

export type AppMetadataForClient = {
    bot_user_id: string;
    bot_username: string;
};

export type AppContext = {
    app_id: string;
    location?: string;
    acting_user_id?: string;
    user_id?: string;
    channel_id?: string;
    team_id?: string;
    post_id?: string;
    root_id?: string;
    props?: AppContextProps;
    user_agent?: string;
    track_as_submit?: boolean;
};

export type AppContextProps = {
    [name: string]: string;
};

export type AppExpandLevel = string;

export type AppExpand = {
    app?: AppExpandLevel;
    acting_user?: AppExpandLevel;
    channel?: AppExpandLevel;
    config?: AppExpandLevel;
    mentioned?: AppExpandLevel;
    parent_post?: AppExpandLevel;
    post?: AppExpandLevel;
    root_post?: AppExpandLevel;
    team?: AppExpandLevel;
    user?: AppExpandLevel;
};

export type AppForm = {
    title?: string;
    header?: string;
    footer?: string;
    icon?: string;
    submit_buttons?: string;
    cancel_button?: boolean;
    submit_on_cancel?: boolean;
    fields?: AppField[];

    // source is used in 2 cases:
    //   - if submit is not set, it is used to fetch the submittable form from
    //     the app.
    //   - if a select field change triggers a refresh, the form is refreshed
    //     from source.
    source?: AppCall;

    // submit is called when one of the submit buttons is pressed, or the
    // command is executed.
    submit?: AppCall;

    depends_on?: string[];
};

export type AppFormValue = string | AppSelectOption | boolean | null;
export type AppFormValues = {[name: string]: AppFormValue};

export type AppSelectOption = {
    label: string;
    value: string;
    icon_data?: string;
};

export type AppFieldType = string;

// This should go in mattermost-redux
export type AppField = {

    // Name is the name of the JSON field to use.
    name: string;
    type: AppFieldType;
    is_required?: boolean;
    readonly?: boolean;

    // Present (default) value of the field
    value?: AppFormValue;

    description?: string;

    label?: string;
    hint?: string;
    position?: number;

    modal_label?: string;

    // Select props
    refresh?: boolean;
    options?: AppSelectOption[];
    multiselect?: boolean;
    lookup?: AppCall;

    // Text props
    subtype?: string;
    min_length?: number;
    max_length?: number;
};

export type UserAutocomplete = {
    users: UserProfile[];

    // out_of_channel contains users that aren't in the given channel. It's only populated when autocompleting users in
    // a given channel ID.
    out_of_channel?: UserProfile[];
};

export type AutocompleteSuggestion = {
    suggestion: string;
    complete?: string;
    description?: string;
    hint?: string;
    iconData?: string;
}

export type AutocompleteSuggestionWithComplete = AutocompleteSuggestion & {
    complete: string;
}

export type AutocompleteElement = AppField;
export type AutocompleteStaticSelect = AutocompleteElement & {
    options: AppSelectOption[];
};

export type AutocompleteDynamicSelect = AutocompleteElement;

export type AutocompleteUserSelect = AutocompleteElement;

export type AutocompleteChannelSelect = AutocompleteElement;

export type FormResponseData = {
    errors?: {
        [field: string]: string;
    };
}

export type AppLookupResponse = {
    items: AppSelectOption[];
}

export type AppCommandFormMap = {[location: string]: AppForm}
