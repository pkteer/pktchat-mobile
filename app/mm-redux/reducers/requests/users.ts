// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {combineReducers} from 'redux';

import {UserTypes} from '@mm-redux/action_types';
import {GenericAction} from '@mm-redux/types/actions';
import {UsersRequestsStatuses, RequestStatusType} from '@mm-redux/types/requests';

import {RequestStatus} from '../../constants';

import {handleRequest, initialRequestState} from './helpers';

function checkMfa(state: RequestStatusType = initialRequestState(), action: GenericAction): RequestStatusType {
    switch (action.type) {
    case UserTypes.CHECK_MFA_REQUEST:
        return {...state, status: RequestStatus.STARTED};

    case UserTypes.CHECK_MFA_SUCCESS:
        return {...state, status: RequestStatus.SUCCESS, error: null};

    case UserTypes.CHECK_MFA_FAILURE:
        return {...state, status: RequestStatus.FAILURE, error: action.error};

    default:
        return state;
    }
}

function login(state: RequestStatusType = initialRequestState(), action: GenericAction): RequestStatusType {
    switch (action.type) {
    case UserTypes.LOGIN_REQUEST:
        return {...state, status: RequestStatus.STARTED};

    case UserTypes.LOGIN_SUCCESS:
        return {...state, status: RequestStatus.SUCCESS, error: null};

    case UserTypes.LOGIN_FAILURE:
        return {...state, status: RequestStatus.FAILURE, error: action.error};

    default:
        return state;
    }
}

function autocompleteUsers(state: RequestStatusType = initialRequestState(), action: GenericAction): RequestStatusType {
    return handleRequest(
        UserTypes.AUTOCOMPLETE_USERS_REQUEST,
        UserTypes.AUTOCOMPLETE_USERS_SUCCESS,
        UserTypes.AUTOCOMPLETE_USERS_FAILURE,
        state,
        action,
    );
}

function updateMe(state: RequestStatusType = initialRequestState(), action: GenericAction): RequestStatusType {
    return handleRequest(
        UserTypes.UPDATE_ME_REQUEST,
        UserTypes.UPDATE_ME_SUCCESS,
        UserTypes.UPDATE_ME_FAILURE,
        state,
        action,
    );
}

export default (combineReducers({
    checkMfa,
    login,
    autocompleteUsers,
    updateMe,
}) as (b: UsersRequestsStatuses, a: GenericAction) => UsersRequestsStatuses);
