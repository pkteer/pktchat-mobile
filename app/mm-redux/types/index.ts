// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import * as calls from '@mmproducts/calls/store/types/calls';

import * as actions from './actions';
import * as bots from './bots';
import * as channels from './channels';
import * as client4 from './client4';
import * as config from './config';
import * as emojis from './emojis';
import * as errors from './errors';
import * as files from './files';
import * as general from './general';
import * as groups from './groups';
import * as integrations from './integrations';
import * as plugins from './plugins';
import * as posts from './posts';
import * as preferences from './preferences';
import * as reactions from './reactions';
import * as requests from './requests';
import * as roles from './roles';
import * as search from './search';
import * as store from './store';
import * as teams from './teams';
import * as threads from './threads';
import * as typing from './typing';
import * as users from './users';
import * as utilities from './utilities';

export {
    config,
    actions,
    bots,
    plugins,
    store,
    calls,
    channels,
    errors,
    emojis,
    files,
    general,
    groups,
    typing,
    integrations,
    utilities,
    posts,
    preferences,
    roles,
    client4,
    search,
    teams,
    requests,
    reactions,
    users,
    threads,
};
