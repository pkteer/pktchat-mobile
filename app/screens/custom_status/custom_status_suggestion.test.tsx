// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {TouchableOpacity} from 'react-native';

import Preferences from '@mm-redux/constants/preferences';
import {CustomStatusDuration} from '@mm-redux/types/users';
import CustomStatusSuggestion from '@screens/custom_status/custom_status_suggestion';
import {shallowWithIntl} from '@test/intl-test-helper';

describe('screens/custom_status_suggestion', () => {
    const baseProps = {
        handleSuggestionClick: jest.fn(),
        emoji: 'calendar',
        text: 'In a meeting',
        theme: Preferences.THEMES.denim,
        separator: false,
        duration: CustomStatusDuration.DONT_CLEAR,
    };

    it('should match snapshot', () => {
        const wrapper = shallowWithIntl(
            <CustomStatusSuggestion
                {...baseProps}
            />,
        );

        expect(wrapper.dive().getElement()).toMatchSnapshot();
    });

    it('should match snapshot with separator and clear button', () => {
        const wrapper = shallowWithIntl(
            <CustomStatusSuggestion
                {...baseProps}
                separator={true}
                handleClear={jest.fn()}
            />,
        );

        expect(wrapper.dive().getElement()).toMatchSnapshot();
    });

    it('should call handleSuggestionClick on clicking the suggestion', () => {
        const wrapper = shallowWithIntl(
            <CustomStatusSuggestion
                {...baseProps}
            />,
        );

        wrapper.dive().find(TouchableOpacity).simulate('press');
        expect(baseProps.handleSuggestionClick).toBeCalled();
    });
});
