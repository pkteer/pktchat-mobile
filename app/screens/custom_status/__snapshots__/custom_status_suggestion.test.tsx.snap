// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screens/custom_status_suggestion should match snapshot 1`] = `
<TouchableOpacity
  onPress={[Function]}
  testID="custom_status_suggestion.In a meeting"
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flexDirection": "row",
        "minHeight": 50,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3f4350",
          "height": 46,
          "left": 14,
          "marginRight": 6,
          "top": 12,
          "width": 45,
        }
      }
    >
      <Connect(Emoji)
        emojiName="calendar"
        size={20}
      />
    </Text>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "paddingBottom": 14,
            "paddingTop": 14,
            "width": "70%",
          }
        }
      >
        <View>
          <CustomStatusText
            text="In a meeting"
            textStyle={
              Object {
                "color": "#3f4350",
              }
            }
            theme={
              Object {
                "awayIndicator": "#ffbc1f",
                "buttonBg": "#1c58d9",
                "buttonColor": "#ffffff",
                "centerChannelBg": "#ffffff",
                "centerChannelColor": "#3f4350",
                "codeTheme": "github",
                "dndIndicator": "#d24b4e",
                "errorTextColor": "#d24b4e",
                "linkColor": "#386fe5",
                "mentionBg": "#ffffff",
                "mentionColor": "#1e325c",
                "mentionHighlightBg": "#ffd470",
                "mentionHighlightLink": "#1b1d22",
                "newMessageSeparator": "#cc8f00",
                "onlineIndicator": "#3db887",
                "sidebarBg": "#1e325c",
                "sidebarHeaderBg": "#192a4d",
                "sidebarHeaderTextColor": "#ffffff",
                "sidebarTeamBarBg": "#14213e",
                "sidebarText": "#ffffff",
                "sidebarTextActiveBorder": "#5d89ea",
                "sidebarTextActiveColor": "#ffffff",
                "sidebarTextHoverBg": "#28427b",
                "sidebarUnreadText": "#ffffff",
                "type": "Denim",
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</TouchableOpacity>
`;

exports[`screens/custom_status_suggestion should match snapshot with separator and clear button 1`] = `
<TouchableOpacity
  onPress={[Function]}
  testID="custom_status_suggestion.In a meeting"
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flexDirection": "row",
        "minHeight": 50,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3f4350",
          "height": 46,
          "left": 14,
          "marginRight": 6,
          "top": 12,
          "width": 45,
        }
      }
    >
      <Connect(Emoji)
        emojiName="calendar"
        size={20}
      />
    </Text>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "paddingBottom": 14,
            "paddingTop": 14,
            "width": "70%",
          }
        }
      >
        <View>
          <CustomStatusText
            text="In a meeting"
            textStyle={
              Object {
                "color": "#3f4350",
              }
            }
            theme={
              Object {
                "awayIndicator": "#ffbc1f",
                "buttonBg": "#1c58d9",
                "buttonColor": "#ffffff",
                "centerChannelBg": "#ffffff",
                "centerChannelColor": "#3f4350",
                "codeTheme": "github",
                "dndIndicator": "#d24b4e",
                "errorTextColor": "#d24b4e",
                "linkColor": "#386fe5",
                "mentionBg": "#ffffff",
                "mentionColor": "#1e325c",
                "mentionHighlightBg": "#ffd470",
                "mentionHighlightLink": "#1b1d22",
                "newMessageSeparator": "#cc8f00",
                "onlineIndicator": "#3db887",
                "sidebarBg": "#1e325c",
                "sidebarHeaderBg": "#192a4d",
                "sidebarHeaderTextColor": "#ffffff",
                "sidebarTeamBarBg": "#14213e",
                "sidebarText": "#ffffff",
                "sidebarTextActiveBorder": "#5d89ea",
                "sidebarTextActiveColor": "#ffffff",
                "sidebarTextHoverBg": "#28427b",
                "sidebarUnreadText": "#ffffff",
                "type": "Denim",
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgba(63,67,80,0.2)",
            "height": 1,
            "marginRight": 16,
          }
        }
      />
    </View>
  </View>
</TouchableOpacity>
`;
